apps/predictions/lib/stream_supervisor.ex

```diff
@@ -24,81 +26,47 @@ defmodule Predictions.StreamSupervisor do
       nil ->
         start_stream(keys)
 
-      pid ->
-        {:ok, pid}
+      stream_pid ->
+        {:ok, stream_pid}
     end
   end
 
   @spec lookup(String.t()) :: pid() | nil
   defp lookup(key) do
-    case Registry.lookup(:prediction_streams_registry, key) do
-      [{_pid, sses_pid}] ->
-        if Process.alive?(sses_pid), do: sses_pid
```

Comment:
question: Is this `Process.alive?` check no longer needed?

My memory is shaky, so this might all be wrong, but here's the situation I thought this was covering: 

A opens a socket and needs a new stream, so we start one and register it. Then B opens a socket for the same route, and we can look up the existing stream in the registry, and B can use that. Then, A and B both close their sockets, so we no longer need the stream, and it can be closed. Then C opens a socket, we go to the registry lookup and find the existing stream pid, but we don't want to actually give that to C because it no longer actually exists.

Am I getting that wrong, and/or is something else covering the duties of what `Process.alive?` was previously doing?

Reply:
In any case, a test for getting a new stream spun up after a previous one was stopped might not be a bad addition.

Reply:
I'm still thinking on this one and will add a test.

My impression is that, now that the stream is wrapped up in a supervisor process that is now tied to the registry via `:via`, that process + registration now live/die together. This change also means the pid being registered/retrieved here is no longer the SSES one directly, but the parent supervisor one.

Now of course, if that child SSES process happens to die for whatever reason, it should be restarted by that supervisor... and I guess if _that_ fails, there are bigger problems to catch here (that maybe I should be checking for here?) 🤔